DOCSP-49976 Update Documents (#517)
mongodb/docs-golang: source/crud/update.txt

* Specify the field and value changes.
* Specify options, if you must modify the method behavior.

The driver provides the following methods to update documents:

* ``UpdateByID()``
* ``UpdateOne()``
* ``UpdateMany()``

Use the ``UpdateOne()`` or ``UpdateByID()`` method to update a single
document.

> Use the ``UpdateMany()`` method to update multiple documents.

Author: Let me know what you think about changes.

field. You cannot use update operations to change the
``_id`` field. If you attempt to change this field, the update
methods return a ``WriteError``.

Sample Data

> ~~~~~~~~~~~

Reviewer: S; It looks like there's no content under this heading, is it missing or is this heading in the wrong place?

.. note:: Aggregation Pipelines in Update Operations

   You can use aggregation pipelines made up of a subset of aggregation 
   stages in update operations. To learn more about 
   the aggregation stages MongoDB supports in

>    aggregation pipelines, see our tutorial on performing
>    :manual:`updates with aggregation pipelines
>    </tutorial/update-documents-with-aggregation-pipeline/>`.

Reviewer: ```suggestion
   aggregation pipelines, see the :manual:`Updates with Aggregation Pipeline
  </tutorial/update-documents-with-aggregation-pipeline/>` tutorial in the Server manual.
```

documents match the query filter, the update operation makes no
changes. 

See the :ref:`upsert guide <golang-upsert-guide>`
to learn how to insert a new document if no documents match the query filter.

> UpdateOne() Example

Author: For technical reviewer: Can you please review this new UpdateOne() Ex? Thank you!

Reviewer 2: 👍 LGTM

> See our :ref:`upsert guide <golang-upsert-guide>`
> to learn how to insert a new document if no documents match the query filter.

Reviewer: S: try to remove first person where possible
```suggestion
See the :ref:`upsert guide <golang-upsert-guide>`
to learn how to insert a new document if no documents match the query filter.
```

      ...

>    }

Author: Updated the ex!

      ...
   }

UpdateMany() Example
--------------------

The following example uses the ``listingsAndReviews`` collection in the 
``sample_airbnb`` dataset from the :atlas:`Atlas sample datasets </sample-data>`.
The example performs the following on the ``listingsAndReviews`` collection:

- Matches documents in which the market field of the address subdocument, ``address.market`` is "Sydney"
- Updates the ``price`` in the matched documents by 1.15 times

.. literalinclude:: /includes/usage-examples/code-snippets/updateMany.go
   :start-after: begin updatemany
   :end-before: end updatemany
   :emphasize-lines: 9
   :language: go
   :copyable:
   :dedent:

The following query updates 609 documents. After you run the example, you can
find the following updated documents in the ``listingsAndReviews`` collection:

.. code-block:: json
   :copyable: false

   // results truncated
   ...
   { "_id" : "10091713", ... , "name" : "Surry Hills Studio", ... , "price" : 181.00, ... },
   { "_id" : "9908871", ... , "name" : "Family friendly beach house", ... , "price" : 751.00, ... },
   { "_id" : "20989061", ... , "name" : "Big and sunny Narraben room", ... , "price" : 60.00, ... },

>    ...

Author: That's fair, I deleted this output. UpdateMany() returns the count of documents updated, so put 609 in monospace to hopefully highlight that a bit more. Let me know what you think!
